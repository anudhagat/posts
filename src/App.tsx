import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import configureStore from './store';
import PostsView from './views/posts-view';
import PostDetailView from './views/post-detail-view';
import UserView from './views/user-view';

const store = configureStore({});

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <BrowserRouter>
          <Route exact path="/" component={PostsView} />
          <Route path="/user" component={UserView} />
          <Route path="/post" component={PostDetailView} />
        </BrowserRouter>
      </Provider>
    );
  }
}

export default App;
