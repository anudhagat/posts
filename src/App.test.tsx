import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('<App />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.debug()).toMatchSnapshot();
    expect(wrapper.find('Provider')).toHaveLength(1);
  });
});
