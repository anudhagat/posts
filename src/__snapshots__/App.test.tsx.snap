// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
"<Provider store={{...}}>
  <BrowserRouter>
    <Route exact={true} path=\\"/\\" component={[Function: PostsView]} />
    <Route path=\\"/user\\" component={[Function: UserView]} />
    <Route path=\\"/post\\" component={[Function: PostDetailView]} />
  </BrowserRouter>
</Provider>"
`;
